pkg/reconciler/jobsink/jobsink.go

```diff
@@ -0,0 +1,177 @@
+/*
+Copyright 2024 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package jobsink
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	"go.uber.org/zap"
+	batchv1 "k8s.io/api/batch/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	batchlisters "k8s.io/client-go/listers/batch/v1"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/utils/ptr"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/network"
+	"knative.dev/pkg/reconciler"
+
+	"knative.dev/eventing/pkg/apis/feature"
+	sinksc "knative.dev/eventing/pkg/apis/sinks"
+	sinks "knative.dev/eventing/pkg/apis/sinks/v1alpha1"
+	"knative.dev/eventing/pkg/auth"
+	"knative.dev/eventing/pkg/eventingtls"
+)
+
+const (
+	maxFailedJobs = 10
```

Comment:
Should this eventually be configurable?

Reply:
I think I want to rethink a solution for this that is more elegant and doesn't require a limit.

I was thinking to show a field `status.job.selector` that you can use to list jobs and failed jobs